common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java

```diff
@@ -134,57 +150,43 @@ protected void onNewIntent(Intent intent) {
     protected void onResume() {
         super.onResume();
 
-        /*
-         * If this is the first run of the activity, start the authorization intent.
-         * If the Authorization Agent is set as WebView. set up the webView client and load the url in embedded webView.
-         * If the Authorization Agent is set as Default or Browser, start the authorization intent with customTabs or browsers.
-         */
-        if (!mAuthorizationStarted) {
-            mAuthorizationStarted = true;
-
-            if (mAuthorizationAgent == AuthorizationAgent.WEBVIEW) {
-                //TODO Replace AzureActiveDirectoryWebViewClient with GenericOAuth2WebViewClient once OAuth2Strategy get integrated.
-                AzureActiveDirectoryWebViewClient webViewClient = new AzureActiveDirectoryWebViewClient(this, new AuthorizationCompletionCallback(), mRedirectUri);
-                setUpWebView(webViewClient);
-                mWebView.post(new Runnable() {
-                    @Override
-                    public void run() {
-                        // load blank first to avoid error for not loading webview
-                        mWebView.loadUrl("about:blank");
-                        Logger.verbose(TAG, "Launching embedded WebView for acquiring auth code.");
-                        Logger.verbosePII(TAG, "The start url is " + mAuthorizationRequestUrl);
-                        mWebView.loadUrl(mAuthorizationRequestUrl);
-                    }
-                });
-            } else {
-                if (mAuthIntent != null) {
-                    startActivity(mAuthIntent);
-                } else {
-                    final Intent resultIntent = new Intent();
-                    resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_AUTHENTICATION_EXCEPTION, new ClientException(ErrorStrings.AUTHORIZATION_INTENT_IS_NULL));
-                    sendResult(AuthorizationStrategy.UIResponse.BROWSER_CODE_AUTHENTICATION_EXCEPTION, resultIntent);
-                    finish();
-                }
-            }
-
+        if (mAuthorizationAgent == AuthorizationAgent.WEBVIEW) {
```

Comment:
Can we remove this side of the conditional and instead gate this logic on...
```java
if (mAuthorizationAgent == AuthorizationAgent.DEFAULT || mAuthorizationAgent.BROWSER) {
    // ...
}
```

 I think this code is easier to understand out-of-context if the logic happens based on what "_is_" rather than "_what isn't_"

Reply:
Addressed. Thanks.